lib/src/kbasesearchengine/main/IndexerWorker.java

```diff
@@ -82,13 +75,14 @@ public IndexerWorker(
             final IndexingStorage indexingStorage,
             final TypeStorage typeStorage,
             final File tempDir,
-            final LineLogger logger) {
+            final LineLogger logger) throws IOException {
         Utils.notNullOrEmpty("id", "id cannot be null or the empty string");
         Utils.nonNull(logger, "logger");
         Utils.nonNull(indexingStorage, "indexingStorage");
         this.id = id;
         this.logger = logger;
-        this.rootTempDir = tempDir;
+        this.rootTempDir = FileUtil.getOrCreateCleanSubDir(tempDir,
+                id+"_"+ UUID.randomUUID().toString().substring(0,5));
```

Comment:
Should probably log the dir name so admins know which dirs are in use

Reply:
Added info log.